packages/epo-widget-lib/src/widgets/ColorTool/hooks/useFilteredImages.tsx

```diff
@@ -0,0 +1,76 @@
+import { useLayoutEffect, useState } from "react";
+import isEqual from "lodash/isEqual";
+import { ImageShape } from "@rubin-epo/epo-react-lib/Image";
+import usePrevious from "@/hooks/usePrevious";
+import { ImageFilter } from "../ColorTool";
+import { getFilters, updateColor } from "../lib/canvas";
+
+interface useFilteredImagesProps {
+  images: Array<ImageShape>;
+  filters: Array<ImageFilter>;
+}
+
+const useFilteredImages = ({
+  images,
+  filters,
+}: useFilteredImagesProps): [Array<HTMLCanvasElement>, boolean] => {
+  const previousImages = usePrevious(images);
+  const [imagesLoaded, setImagesLoaded] = useState(0);
+  const [imgElements, setImages] = useState<Array<HTMLImageElement>>();
+
+  const anyImageIncomplete = imgElements?.some(({ complete }) => !complete);
+
+  const loading = imagesLoaded !== images.length && anyImageIncomplete;
+
+  useLayoutEffect(() => {
+    if (!isEqual(previousImages, images)) {
+      setImagesLoaded(0);
+      setImages(
+        images.map(({ url, width, height }) => {
+          const img = new Image(width, height);
+          img.onload = () => {
+            setImagesLoaded((value) => value + 1);
+          };
+          img.src = url;
+
+          return img;
+        })
+      );
+    }
+  }, [images]);
+
+  if (loading) {
```

Comment:
we determine loading status for all of the items, and return nothing while loading is happening